### PR TITLE
Added Player.isMoving() and a small bug fix

### PR DESCRIPTION
**Pull request by realfork**
Convenience method for checking if a player is moving

### CODE REVIEW

**Comment by realfork** [comment]
**Changes:**

- RenderEntityModel trigger
- EntityCull trigger (Unsure about it, can be deleted)
- MotionUpdatePre + MotionUpdatePost (Not 100% sure what to name these since it isn't technically motion (onUpdateWalkingPlayer))
- Fix to RenderTitle trigger

**Comment by mattco98** [comment]
Can you provide use cases for these? For example, if I understand it correctly, the movement trigger can easily be achieved with a tick trigger and an if statement, and I'm not sure what you would use the pre/post versions for, since they are not cancellable. And it seems like instead of adding triggers for model parts, it would be much more useful to add accessors for them to the various entity classes that could be used in the `renderEntity` trigger that already exists.

**Comment by realfork** [comment]
Most of them are for convenience.

RenderEntityModel - ModelBase is painful to get even with accessors so it's to massively simplify that process.
Movement - Is nicer than writing a tick trigger with the function but it can be removed.
MotionUpdate triggers - For running code before and after movement packets are sent.

**Comment by mattco98** [comment]
> RenderEntityModel - ModelBase is painful to get even with accessors so it's to massively simplify that process.

Right, which is why CT can potentially take that burden off of the users with accessors, even if they are "painful". If it is feasible, we should go that route instead. And the bar for feasible for CT code (.kt) is much much lower than it is for user modules. 

> Movement - Is nicer than writing a tick trigger with the function but it can be removed.

I agree with the removal, doesn't seem to have much motivation

> MotionUpdate triggers - For running code before and after movement packets are sent.

That's not a use case, that is what they do, which I can infer from the code. What do you plan on doing with them that cannot be done another way?

**Comment by realfork** [comment]
Ok, I finally found the way to get main model so I added it to the entity wrapper and removed the trigger. I also removed the movement triggers.

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt [hunk 1]
```diff
@@ -151,6 +154,13 @@ open class Entity(val entity: MCEntity) {
      */
     fun getClassName(): String = entity.javaClass.simpleName
 
+    /**
+     * Gets the ModelBase of the entity
+     *
+     * @return the entity's main model
+     */
+    fun getMainModel(): ModelBase = (Client.getMinecraft().renderManager.getEntityRenderObject<net.minecraft.entity.Entity>(entity) as RendererLivingEntity).getMainModel()
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt]: Can this cast fail?

Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt]: We probably want a simple wrapper around `ModelBase` which forwards methods to an underlying implementation so the user doesn't need to use obfuscated methods. See [`Particle`](https://github.com/ChatTriggers/ChatTriggers/blob/master/src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Particle.kt) for an example. Idk what `ModelBase` looks like but as in `Particle`, feel free to change method names or add helper methods if you want.

Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt]: We should have an `MCEntity` alias in [`TypeAliases.kt`](https://github.com/ChatTriggers/ChatTriggers/blob/master/src/main/kotlin/com/chattriggers/ctjs/utils/kotlin/TypeAliases.kt) so you don't have to write out the full package name

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/Player.kt [hunk 2]
```diff
@@ -195,6 +195,9 @@ object Player {
     @JvmStatic
     fun getLightLevel(): Int = World.getWorld()?.getLight(getPlayer()?.position) ?: 0
 
+    @JvmStatic
+    fun isMoving(): Boolean = getPlayer()?.let { it.movementInput.moveForward != 0F || it.movementInput.moveStrafe != 0F } ?: false
```
Comment (mattco98) [inline src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/Player.kt]: Small nitpick to shorten the line a bit:
```suggestion
    fun isMoving(): Boolean = getPlayer()?.movementInput?.let { it.moveForward != 0F || it.moveStrafe != 0F } ?: false
```

**Review by mattco98 — CHANGES_REQUESTED** [review]
Looks much better, thanks for refactoring!

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt (hunk 1 above)
Comment (realfork) [reply src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt]: I don't think so?

**Comment by realfork** [comment]
Sorry for abandoning this for a while, I removed the model base method since it doesn't actually get the model correctly. Should be ready now.

**Pull request merged**